src/term.rs

```diff
@@ -408,9 +545,10 @@ impl Term {
                 func(t1);
                 func(t2);
             }
-            OpN(_, ref mut terms) | Array(ref mut terms, _) => terms.iter_mut().for_each(|t| {
+            OpN(_, ref mut terms) => terms.iter_mut().for_each(|t| {
                 func(t);
             }),
+            Array(ref mut terms, _) => terms.get_mut().unwrap().iter_mut().for_each(|t| func(t)),
```

Comment:
`Rc::make_mut` doesn't typecheck when calling it on `Rc<[T]>` because "`[T]` doesn't implement `Clone`"
I was able to work around this and left a comment.